torch/serialization.py

```diff
@@ -453,14 +453,31 @@ def persistent_load(saved_id):
             raise RuntimeError("Unknown saved id type: %s" % saved_id[0])
 
     f_is_real_file = _is_real_file(f)
-    if f_is_real_file and f.tell() == 0:
+
+    def raise_load_err_msg(pattern, e):
+        if pattern in str(e):
+            msg = (str(e) + ". You can only torch.load from a file that is seekable." +
+                            " Please pre-load the data into a buffer like io.BytesIO and try to load from it instead.")
+            raise type(e)(msg)
+        else:
+            raise e
+
+    try:
+        f_tell_is_0 = True if f.tell() == 0 else False
+    except AttributeError as e:
+        raise_load_err_msg("instance has no attribute 'tell'", e)
+
```

Comment:
Instead of wrapping the following `f.seek(0)` call in a try/except, I think it might be better to do something like
```
try:
    f.seek(f.tell())  # should be a no-op, but tells us if seek is implemented
except:...
```
right after you figure out if `f.tell()` is supported or not.

This is because seek is called elsewhere in the load logic, and we might not always enter this particular pathway. In particular, the following two approaches could work:
1) do a try/catch at every `f.seek` site.
2) do a try/catch at the beginning of the load logic so we're certain that `f.seek` is supported. [what I'm suggesting here]

Reply:
This is a great point! I totally agreed! I am also thinking about to do a check similar to [_should_read_directly](https://github.com/weiyangfb/pytorch/blob/c25f165e3c2be7f22e8d65dbb8f8a1a14b351ee9/torch/serialization.py#L495). Maybe I can write a  method called `can_seek_and_tell()`, and do `try..catch` from there?

Reply:
go for it!